core/model/src/geometry_sampled_field.cpp

```diff
@@ -172,44 +186,66 @@ static std::vector<QRgb> setImagePixels(
   } else {
     values = common::stringToVector<T>(sampledField->getSamples());
   }
-  SPDLOG_ERROR("Importing sampled field of {} samples of type {}",
+  SPDLOG_DEBUG("Importing sampled field of {} samples of type {}",
                values.size(), common::decltypeStr<T>());
-  if (static_cast<int>(values.size()) != sampledField->getSamplesLength()) {
-    SPDLOG_WARN("Number of samples {} doesn't match SamplesLength {}",
-                values.size(), sampledField->getSamplesLength());
-  }
-  std::size_t iCol = 0;
-  auto iter = colours.begin();
-  for (const auto *sampledVolume : sampledVolumes) {
-    auto matches = getMatchingSampledValues(values, sampledVolume);
-    if (std::find(matches.cbegin(), matches.cend(), true) != matches.cend()) {
-      auto col = common::indexedColours()[iCol].rgb();
-      SPDLOG_DEBUG("  {}/{} -> colour {:x}", sampledVolume->getId(),
-                   sampledVolume->getDomainType(), col);
-      setMatchingPixelsToColour(img, matches, col);
+  if (common::isItIndexes(values, importedSampledFieldColours.size())) {
+    setImportedColoursToImage(img, values, importedSampledFieldColours);
+    auto iter = colours.begin();
+    for (const auto *sampledVolume : sampledVolumes) {
+      int sampledValue = sampledVolume->getSampledValue();
+      auto col = importedSampledFieldColours[sampledValue];
+      SPDLOG_DEBUG("importedSampledFieldColours  {}/{} -> colour {:x}",
+                   sampledVolume->getId(), sampledVolume->getDomainType(), col);
       *iter = col;
-      ++iCol;
+      ++iter;
+    }
```

Comment:
`col` here is the pixel colour that will be used to identify which pixels are in the compartment - so if `importedSampledFieldColours` are wrong (e.g. correct values but in the wrong order) the compartments will have the wrong set of pixels. (That would be a good test case to have by the way)
We want the model geometry to be correct regardless of what is in `importedSampledFieldColours` - if we remove this code and the `else` and uncomment the commented out lines below then I think we should always get the right geometry. (although we do a bit of unnecessary work if the colours were correct since we are again assigning the pixel colours for the sampled volumes)